RecoTracker/FinalTrackSelectors/plugins/TrackSelectorByRegion.cc

```diff
@@ -0,0 +1,85 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/Common/interface/OwnVector.h"
+#include "RecoTracker/TkTrackingRegions/interface/TrackingRegion.h"
+#include "RecoTracker/TkTrackingRegions/interface/RectangularEtaPhiTrackingRegion.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include <vector>
+#include <memory>
+
+namespace {
+  class TrackSelectorByRegion final : public edm::global::EDProducer<> {
+  public:
+    explicit TrackSelectorByRegion(const edm::ParameterSet& conf)
+        : tracksToken_(consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks"))) {
+      inputTrkRegionToken_ = consumes<edm::OwnVector<TrackingRegion>>(conf.getParameter<edm::InputTag>("inputROI"));
+      produce_collection = conf.getParameter<bool>("produceTrackCollection");
```

Comment:
as briefly discussed today, it might be useful to have an other boolean parameter for handling the mask collection